_posts/en/newsletters/2024-02-07-newsletter.md

```diff
@@ -0,0 +1,319 @@
+    connections can delay a Bitcoin node from receiving a block for
+    roughly the time it takes to produce _x_ blocks.  If the forwarding
+    payment has to be claimed within 40 blocks, an attacker controlling
+    50 connections can have a reasonable chance of preventing the
+    Bitcoin node from seeing the block containing the preimage until the
+    spending node is able to receive a refund of the payment.  If that
+    happens, the attacker's spending node paid nothing and the
+    attacker's receiving node received an amount extracted from the
+    victim's node.
+
+    As Siegel reports, two changes were made to Bitcoin Core to prevent
+    stalling:
+
+    - [Bitcoin Core #22144][] randomizes the order in which peers are
+      serviced in the message handling thread.  See [Newsletter
+      #154][news154 stall].
+
+    - [Bitcoin Core #22147][] keeps at least one outbound high bandwidth
+      compact block peer even if inbound peers seem to be performing
+      better.  The local node selects its own outbound peers, meaning
+      they're less likely to be under the control of an attacker, so
+      it's useful to keep at least one outbound peer for safety.
+
+- **Securely opening zero-conf channels with v3 transactions:**
+  Matt Corallo [posted][corallo 0conf] to Delving Bitcoin to discuss how
+  to securely allow [zero-conf channel opening][topic zero-conf
+  channels] when the proposed [v3 transaction relay policy][topic v3
+  transaction relay] is being used.  Zero-conf channel opens are new
+  single-funded channels where the funder gives some or all of their
+  initial funds to the acceptor.  Those funds are not secure until the
+  channel open transaction receives a sufficient number of
+  confirmations, so there’s no risk to the acceptor spending some of
+  those funds back through the funder using the standard LN protocol.
+  The initial proposal for v3 transaction relay policy would only allow
+  an unconfirmed v3 transaction to have, at most, a single child in the
+  mempool; the expectation is that the single child will [CPFP fee
+  bump][topic cpfp] its parent if necessary.
+
+    Those v3 rules are incompatible with both parties being able to fee
+    bump a zero-conf channel open: the funding transaction which creates
+    the channel is the parent of a v3 transaction which closes the
+    channel and the grandparent of a v3 transaction for fee bumping.
+    Since the v3 rules only allow one parent and one child, there's no
+    way for the funding transaction to be fee bumped without modifying
+    how it is created.  Bastien Teinturier [notes][teinturier splice]
+    that [splicing][topic splicing] encounters a similar problem.
+
+    As of this writing, the main solution proposed appeared to be
+    modifying funding and splicing transactions to include an extra
+    output for CPFP fee bumping now, wait for [cluster mempool][topic
+    cluster mempool] to hopefully allow v3 to permit more permissive
```

Comment:
```suggestion
    cluster mempool] to allow v3 to permit more permissive
```

Reply:
Leaving as-is. I don't think we can say for certain that cluster mempool will make that possible.

Reply:
That's fine, I think readers will understand what is meant, but I remember reading a long time ago that "hopefully" is often used incorrectly, and shouldn't be used at all in many cases. I can't recall the details, but I think the problem is (using this as an example), "hopefully" appears to be modifying "allow", which is not the intended meaning. To me, it reads as imprecise language (but maybe it's just me). Maybe something like:
```
As of this writing, the main solution proposed appeared to be
modifying funding and splicing transactions to include an extra
output for CPFP fee bumping now, in anticipation of cluster mempool
allowing v3 to permit more permissive topologies ...
```
(I think it's understood that anticipated events might never occur. -- especially in bitcoin!)